[Playlists] Add manual playlist episodes events

bjtitus opened the pull request:
Fixes [PCIOS-247](https://linear.app/a8c/issue/PCIOS-247/playlist-analytics-add-episodes-to-playlist)

Adds the following events:
* `filter_add_episodes_shown`
* `filter_add_episodes_folder_tapped`
* `filter_add_episodes_podcast_tapped`
* `filter_add_episdoes_episode_tapped`

Also the toast for the maximum playlist limit:
<img width="300" alt="CleanShot 2025-10-29 at 21 47 02@2x" src="https://github.com/user-attachments/assets/ee98b292-3b6a-46a5-90d1-e4ccdd9e67cc" />

## To test

- Enable `tracksLogging` and `playlistsRebranding` feature flags
- Create a new Manual Playlist
- Tap “Add Episodes”
- ✅ Make sure you see:

```
🔵 Tracked: filter_add_episodes_shown [:]
```

- Tap a Folder
- ✅ Make sure you see:

```
🔵 Tracked: filter_add_episodes_folder_tapped [:]
```

- Tap a Podcast from the list
- ✅ Make sure you see:

```
🔵 Tracked: filter_add_episodes_podcast_tapped [:]
```

- Tap the Add to Playlist Plus button on an episode
- ✅ Make sure you see:

```
🔵 Tracked: filter_add_episodes_episode_tapped ["is_playlist_full": false]
```

- Set `EpisodeDataManager.Constants.Limits.maxPlaylistItems` to `2`
- Add an episode to a manual playlist
- ✅ Verify you see a toast saying “Playlist is full. Try creating a new one” and the following logged:

```
🔵 Tracked: filter_add_episodes_episode_tapped ["is_playlist_full": true]
```

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

Commit:
Add playlist episode events analytics tracking

- Add filterAddEpisodesShown event when add episodes view appears
- Add filterAddEpisodesFolderTapped event for folder selection
- Add filterAddEpisodesPodcastTapped event for podcast selection
- Add filterAddEpisodesEpisodeTapped event for episode addition
- Add missing analytics events for playlist actions (archive, reorder, episode detail)
- Add filterShown event when playlist detail view appears

bjtitus committed:
Add Toast and isFull logging for is_playlist_full

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

bjtitus committed:
Update full playlist message

The other didn't really make sense without a period at the end and this matches the Android version.
https://github.com/Automattic/pocket-casts-android/pull/4446

bjtitus committed:
Use allEpisodeCount for playlist limit check

danielebogo commented on podcasts/en.lproj/Localizable.strings:
```diff
@@ -5556,7 +5556,7 @@
 
 /* Toast message when the playlist is full */
-"playlist_manual_add_episode_full_playlist_toast" = "Playlist is full. Try creating a new one";
+"playlist_manual_add_episode_full_playlist_toast" = "This playlist is full. Remove a few episodes or start a new one.";
```
We need to use a different text in this scenario. `Playlist is full. Try creating a new one` is used when adding a single episode to manual playlists from the playlists chooser.

danielebogo replied:
| Add multiple episodes | Add single episode |
| :-------: | :-------: |
| <img width="523" height="105" alt="Screenshot 2025-10-30 at 09 40 46" src="https://github.com/user-attachments/assets/aa27e578-1a39-4959-a2e4-da7d6f3b8ea1" /> | <img width="631" height="125" alt="Screenshot 2025-10-30 at 09 40 26" src="https://github.com/user-attachments/assets/919b27ea-8f35-4c5d-9711-f7bcca0d431a" /> |

danielebogo commented on podcasts/New Detail/Episode Add/ManualPlaylistsChooserViewController.swift:
```diff
@@ -152,7 +152,11 @@ class ManualPlaylistsChooserViewController: PCViewController {
-                dataManager.add(episodes: [episode], to: playlist)
+                let didAdd = dataManager.add(episodes: [episode], to: playlist)
+                guard didAdd else {
+                    Toast.show(L10n.playlistManualAddEpisodeFullPlaylistToast)
```
We should use a `ToastTheme` to inject and set the left icon
<img width="523" height="105" alt="Screenshot 2025-10-30 at 09 40 46" src="https://github.com/user-attachments/assets/807fb39e-8bc2-42af-9d88-37fae549e4e9" />

danielebogo reviewed (changes requested):
All good @bjtitus

Events tracked correctly
```
🔵 Tracked: filter_add_episodes_shown [:]
🔵 Tracked: filter_add_episodes_folder_tapped [:]
🔵 Tracked: filter_add_episodes_podcast_tapped [:]
🔵 Tracked: filter_add_episodes_episode_tapped ["is_playlist_full": false]
🔵 Tracked: filter_add_episodes_episode_tapped ["is_playlist_full": true]
```

The only thing I saw is the text to be used in the toast message in this case, which is different from the one used in the Playlists chooser, and the missing left icon in the toast.

bjtitus committed:
Add left icon to all maximum playlist alerts

bjtitus commented:
@danielebogo Ready to go!

danielebogo committed:
Fix test override naming

danielebogo reviewed (approved):
All good! Thanks for addressing the changes!
